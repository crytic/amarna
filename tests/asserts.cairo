func check_price_signature(range_check_ptr) -> (range_check_ptr: felt) {
    // Check ranges.
    assert_nn_le{range_check_ptr=range_check_ptr}(sig.external_price, EXTERNAL_PRICE_UPPER_BOUND);
    assert_nn_le{range_check_ptr=range_check_ptr}(sig.timestamp, TIMESTAMP_BOUND);
    assert_nn_le{range_check_ptr=range_check_ptr}(trade.actual_b_fee, AMOUNT_UPPER_BOUND - 1);
    assert_nn_le{range_check_ptr=range_check_ptr}(trade.actual_b_fee, AMOUNT_UPPER_BOUND - 2);
    assert_nn_le{range_check_ptr=range_check_ptr}(trade.actual_b_fee, AMOUNT_UPPER_BOUND);
}
