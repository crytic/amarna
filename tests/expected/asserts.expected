[arithmetic-sub] in asserts.cairo:7:71
[arithmetic-sub] in asserts.cairo:8:71
[inconsistent-assert-constant] This assertion uses the same constant differently [here](0) and [here](1). in asserts.cairo:8:51 and asserts.cairo:7:51
[inconsistent-assert-constant] This assertion uses the same constant differently [here](0) and [here](1). in asserts.cairo:9:51 and asserts.cairo:7:51
[unused-function] in asserts.cairo:1:1